index.js

```diff
@@ -198,14 +212,17 @@ URL.prototype.set = function set(part, value, fn) {
     url[part] = value;
 
     if (!required(value, url.protocol)) {
-      url.host = url.hostname;
+      url.host = isIPv6(url.hostname) ? '['+ url.hostname +']' : url.hostname;
       url[part] = '';
     } else if (value) {
-      url.host = url.hostname +':'+ value;
+      url.host = isIPv6(url.hostname) ? '['+ url.hostname +']' +':'+ value : url.hostname +':'+ value;
```

Comment:
I duplicated efforts here to avoid making a temporary variable, but that was just a guess, a temp variable would clean it up.

```js
var tmpHost = isIPv6(url.hostname) ? '['+ url.hostname +']' : url.hostname;
if (!required(value, url.protocol)) {
  url.host = tmpHost;
  url[part] = '';
} else if (value) {
  url.host = url.hostname +':'+ value;
  url.host = tmpHost +':'+ value;
}